llvm/lib/Support/Regex.cpp

```diff
@@ -82,8 +82,8 @@ unsigned Regex::getNumMatches() const {
 
 bool Regex::match(StringRef String, SmallVectorImpl<StringRef> *Matches,
                   std::string *Error) const {
-  // Exit without match if string is empty
-  if (String.empty())
+  // No match for null string data
+  if (String.data() == nullptr)
```

Comment:
Now can we have a non-empty string with nullptr data?

Reply:
Pretty sure this indicates a bug in some caller. Possibly the StringRef Data + Length constructor should be asserting that if Length != 0 then also Data != nullptr.

Reply:
Ah sorry, it looks like I was looking at the diff from only the last commit and thought the empty() check is pre-existing and you're changing it to check nullptr instead.

I believe what you want to do is below where String.data() is used, if it it nullptr pass "" instead. That should handle cases where the regex matches an empty string without passing nullptr into the implementation.

Reply:
Thanks! Updating commit.